homework5/create_api_module.py

```diff
@@ -0,0 +1,316 @@
+        user_data = [{'user_full_name': row['Name'], 'birth_day': row['Birth_day'], 'accounts': row['Accounts']}
+                     for row in reader]
+        return add_user(user_data)
+
+
+def add_banks_from_csv(path):
+    """
+    Add bank data from a CSV file to the database.
+    Args:
+        path (str): The path to the CSV file containing bank data.
+    """
+    with open(path, 'r') as csvfile:
+        reader = list(csv.DictReader(csvfile))
+        bank_data = [{'name': row['Bank'], 'id': row['Id']} for row in reader]
+        return add_bank(bank_data)
+
+
+def add_accounts_from_csv(path):
+    """
+    Add account data from a CSV file to the database.
+    Args:
+        path (str): The path to the CSV file containing account data.
+    """
+    with open(path, 'r') as csvfile:
+        reader = list(csv.DictReader(csvfile))
+        account_data = [{'user_id': row['User_id'], 'type': row['Type'], 'account_number': row['Account Number'],
+                         'bank_id': row['Bank_id'], 'currency': row['Currency'], 'amount': row['Amount'],
+                         'status': row['Status']} for row in reader]
+        return add_account(account_data)
+
+
+@db_connection_decorator
+def modify_row(cursor, table, user_id=None, bank_id=None, account_id=None, new_data=None):
+    """
+    Modify a row in the specified table based on the provided parameters.
+    Args:
+        cursor (sqlite3.Cursor): The SQLite database cursor.
+        table (str): The name of the table to modify.
+        user_id (int, optional): The ID of the user to modify (default: None).
+        bank_id (int, optional): The ID of the bank to modify (default: None).
+        account_id (int, optional): The ID of the account to modify (default: None).
+        new_data (dict): A dictionary containing the new data to update.
+    """
+    if not new_data:
+        my_logger.warning('No new data provided for update')
+        return
+
+    id_column = 'Id'
+
+    query_parts = [f' {key} = ?' for key in new_data.keys()]
+    query = f'UPDATE {table} SET' + ','.join(query_parts) + f' WHERE {id_column} = ?'
+    params = list(new_data.values())
+    params.append(user_id if user_id else bank_id if bank_id else account_id)
+
+    cursor.execute(query, params)
+    my_logger.info(f'Row in {table} modified successfully')
+
+
+@db_connection_decorator
+def delete_row(cursor, table, row_id):
+    """
+    Delete a row from the specified table based on the provided row ID.
+    Args:
+        cursor (sqlite3.Cursor): The SQLite database cursor.
+        table (str): The name of the table to delete from.
+        row_id (int): The ID of the row to delete.
+    """
+    cursor.execute(f'SELECT * FROM {table} WHERE Id = ?', (row_id,))
+    existing_row = cursor.fetchone()
+
+    if not existing_row:
+        my_logger.warning(f'Row with Id {row_id} not found in {table}')
+        return
+
+    cursor.execute(f'DELETE FROM {table} WHERE Id = ?', (row_id,))
```

Comment:
I think for such purpose should exist kind of DELETE IF EXIST functionality. please google it